numpy/core/tests/test_multiarray.py

```diff
@@ -1812,30 +1800,21 @@ def test_sort(self):
             c.sort(kind=kind)
             assert_equal(c, a, msg)
 
-        # test datetime64 sorts.
-        a = np.arange(0, 101, dtype='datetime64[D]')
+    @pytest.mark.parametrize('message, dtype', [('datetime64', 'datetime64[D]'), ('timedelta64', 'timedelta64[D]')])
+    def test_sort_time(self, message, dtype):
+        # test datetime64 and timedelta64 sorts.
+        a = np.arange(0, 101, dtype=dtype)
         b = a[::-1]
         for kind in ['q', 'h', 'm']:
-            msg = "datetime64 sort, kind=%s" % kind
-            c = a.copy()
-            c.sort(kind=kind)
-            assert_equal(c, a, msg)
-            c = b.copy()
-            c.sort(kind=kind)
-            assert_equal(c, a, msg)
-
-        # test timedelta64 sorts.
-        a = np.arange(0, 101, dtype='timedelta64[D]')
-        b = a[::-1]
-        for kind in ['q', 'h', 'm']:
-            msg = "timedelta64 sort, kind=%s" % kind
+            msg = message + "sort, kind=%s" % kind
```

Comment:
No need to include this in the message, it's now in the test name thanks to `parametrize`